odoo/tools/query.py

```diff
@@ -272,3 +303,55 @@ def __len__(self) -> int:
 
     def __iter__(self) -> Iterator[int]:
         return iter(self.get_result_ids())
+
+    @property
+    def get(self) -> IdentifierBuilder:
+        # Mapping could be saved on the registry
+        model = next(model._name for model in self._env.registry.values() if model._table == self.table)
+        return IdentifierBuilder(self, self.table, model, 'id')
+
+
+class IdentifierBuilder(SQLable):
+    def __init__(self, query: Query, alias: str, model: str, fname :str):
+        self.query = query
+        self.env = query._env
+        self.alias = alias
+        self.model = model
+        self.fname = fname
+        self._join_kind = None
+
+    def __getattr__(self, name: str, /) -> IdentifierBuilder:
+        alias, model = self.alias, self.model
+        field = self.env[self.model]._fields[self.fname]
+        if field.type == 'many2one':
+            alias = self.join(self.alias, self.fname, self.env[field.comodel_name]._table, 'id', self.fname)
+            model = field.comodel_name
+        elif field.type == 'one2many':
```

Comment:
If x2m is magically joining it is error prone. Sometime you want an EXiSTS or IN or some JOIN - that should be manual.